File: number_divisible_by_digits.rb
# http://codegolf.stackexchange.com/questions/41902/is-a-number-divisible-by-each-of-its-digits

def divisible_by_each_digit?(number)
  number.to_s.split(//).map { |digit| digit.to_i }.each do |digit|
    return false if digit == 0 || number % digit != 0
  end
  true
end

p divisible_by_each_digit?(120)
p divisible_by_each_digit?(128)

File: number_divisible_by_digits.swift
#!/usr/bin/env swift
// http://codegolf.stackexchange.com/questions/41902/is-a-number-divisible-by-each-of-its-digits

func divisibleByEachDigit(n: Int) -> Bool {
  let digits = String(n).characters.map { Int(String($0))! }
  for d in digits {
    if d == 0 || n % d != 0 { return false }
  }
  return true
}

print(divisibleByEachDigit(120))
print(divisibleByEachDigit(128))
